test/eigensolver/lobpcg_kernels.cpp

```diff
@@ -0,0 +1,196 @@
+// SPDX-FileCopyrightText: 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/eigensolver/lobpcg_kernels.hpp"
+
+#include <algorithm>
+#include <limits>
+
+#include <gtest/gtest.h>
+
+#include <ginkgo/core/base/array.hpp>
+#include <ginkgo/core/base/exception.hpp>
+#include <ginkgo/core/base/executor.hpp>
+#include <ginkgo/core/base/math.hpp>
+#include <ginkgo/core/matrix/dense.hpp>
+
+#include "core/test/utils.hpp"
+#include "test/utils/common_fixture.hpp"
+
+
+template <typename ValueType>
+class Lobpcg : public CommonTestFixture {
+protected:
+    using rc_value_type = gko::remove_complex<value_type>;
+    using cmplx_value_type = gko::to_complex<rc_value_type>;
+    using Mtx_r = gko::matrix::Dense<rc_value_type>;
+    using Mtx_c = gko::to_complex<Mtx_r>;
+    using Mtx = gko::matrix::Dense<value_type>;
+    using Ary_r = gko::array<rc_value_type>;
+    using Ary = gko::array<value_type>;
+    Lobpcg()
+    {
+        small_a_r = gko::initialize<Mtx_r>({{13.2, -4.3, -1.8, 0.12},
+                                            {-4.3, 24.2, -1.7, -2.3},
+                                            {-1.8, -1.7, 18.7, -0.8},
+                                            {0.12, -2.3, -0.8, 10.0}},
+                                           ref);
+        small_b_r = gko::initialize<Mtx_r>({{2.0, -1.1, 0.3, 0.01},
+                                            {-1.1, 2.5, -0.8, 0.5},
+                                            {0.3, -0.8, 2.3, -0.2},
+                                            {0.01, 0.5, -0.2, 1.9}},
+                                           ref);
+        small_a_cmplx = gko::initialize<Mtx_c>(
```

Comment:
suggest to split the complex type to different one.
We have ComplexValueTypesBase (if no half) for that.
because the interface between complex and real  are the same, we test the small_a_r, small_b_r with all value types.
and small_a_cmplx, small_b_complx with ComplexValueType

Reply:
What do you mean by "different one"?  I don't think I understand.